docs/github_pages_version_file.py

```diff
@@ -0,0 +1,175 @@
+) -> str:
+    """Create the string for the RST file."""
+    file_str = ""
+    # Add branches first
+    for build in main_branches:
+        file_str += f"{build.version} <{build.url}>\n"
+    for build in branches:
+        file_str += f"{build.version} <{build.url}>\n"
+    # Add releases
+    for build in releases:
+        file_str += f"{build.version} <{build.url}>\n"
+    return file_str
+
+
+def push_file_to_GitHub(file_string: str, filename: str, remote: str, branch: str):
+    """Push the version file to the GitHub Pages branch."""
+    # Check out a local branch tracking the GitHub Pages branch
+    run_command(f"git checkout -tq {remote}/{branch}")
+
+    try:
+        # Write the file
+        with open(filename, "w") as output_file:
+            output_file.write(file_string)
+
+        # Check if file is already tracked
+        status = run_command(
+            f"git ls-files --error-unmatch {filename}",
+            return_exit_status=True,
+            fail_on_bad_status=False,
+        )
+
+        # Status of zero means tracked. So check if file has changed
+        if not status:
+            status = run_command(
+                f"git diff --exit-code {filename}", return_exit_status=True
+            )
+
+        # If status is one (either untracked or file has changed) then commit the file
+        if status:
+            print(f"File {filename} being commited to {remote}/{branch}.")
+            # Add the file
+            run_command(f"git add {filename}")
+
+            # Commit the file
+            run_command("git commit -qm Update-version-file")
+
+            # Push to remote
+            run_command("git push -q")
+        else:
+            print(f"File {filename} has not been updated. Not committing.")
+
+    finally:
+        # Revert back to original branch
+        run_command("git checkout -q -")
+
+        # Delete local branch
+        run_command(f"git branch -q -D {branch}")
+
+
+def create_gh_pages_versions_file():
+    """Generate the file containing the list of all GitHub Pages builds."""
+    # General information
+    github_org = "dls-controls"
+    remote = "origin"
+    # TODO: replace branch with actual gh-pages branch
+    github_pages_branch = "test/gh-pages-file"
+    project = conf.project
+    version_filename = "version.txt"
+
+    # Store the builds into different groups
+    main_branches: List[GitHubPagesBuild] = []
+    releases: List[GitHubPagesBuild] = []
+    branches: List[GitHubPagesBuild] = []
+
+    # Get the list of sorted tags
+    tags = get_sorted_tags_list()
+
+    # Get the directories (i.e. builds) from the GitHub Pages branch
+    all_build_versions = get_branch_contents(remote, github_pages_branch)
+
+    # Parse all builds, storing releases in an unordered dict
+    unsorted_releases: Dict[str, GitHubPagesBuild] = {}
+    for version in all_build_versions:
+        # Create our instance
+        url = get_docs_url(github_org, project, version)
```

Comment:
Don't need the URL, just the name

Reply:
Now just creates a text file of a list of versions: https://github.com/dls-controls/scanspec/blob/test/gh-pages-file/version.txt